### ISSUE TITLE
Add timePart extractionFn (Or make timeFormat more useful)

### ISSUE THREAD

**Issue by vogievetsky** [apache/incubator-druid#1877]
In Plywood/PlyQL, to compute:

``` sql
SELECT TIME_PART(`time`, SECOND_OF_HOUR, 'Etc/UTC') AS 'SecondOfHour' FROM `wiki` GROUP BY 1
```

I issue a timeFormat with the format `m'*60+'s` to get values like `1*60+13` that I then compute it into `73`.
Want to see more crazy hack? https://github.com/implydata/plywood/blob/master/src/external/druidExternal.ts#L7

I would love it if there was some sensible way to get the values for

```
SECOND_OF_MINUTE
SECOND_OF_HOUR
SECOND_OF_DAY
SECOND_OF_WEEK
SECOND_OF_MONTH
SECOND_OF_YEAR

MINUTE_OF_HOUR
MINUTE_OF_DAY
MINUTE_OF_WEEK
MINUTE_OF_MONTH
MINUTE_OF_YEAR

HOUR_OF_DAY
HOUR_OF_WEEK
HOUR_OF_MONTH
HOUR_OF_YEAR

DAY_OF_WEEK
DAY_OF_MONTH
DAY_OF_YEAR

WEEK_OF_MONTH
WEEK_OF_YEAR
```

I am picturing something like:

``` json
{
  "type" : "extraction",
  "dimension" : "__time",
  "outputName" :  "SecondOfHour",
  "extractionFn" : {
    "type" : "timePart",
    "part" : "SECOND_OF_HOUR",
    "timeZone" : "America/Montreal"
  }
}
```

**Comment by xvrl** [comment]
this already exists, RTFM :) http://druid.io/docs/latest/querying/dimensionspecs.html#time-format-extraction-function

**Comment by vogievetsky** [comment]
Just to be clear, as you can see here: https://github.com/implydata/plywood/blob/master/src/external/druidExternal.ts#L7 

it is pretty easy to do this hacky time parting for:

SECOND_OF_MINUTE: "s",
MINUTE_OF_HOUR: "m",
HOUR_OF_DAY: "H",
DAY_OF_WEEK: "e",
DAY_OF_MONTH: "d",
DAY_OF_YEAR: "D",
WEEK_OF_YEAR: "w"

Just because there happen to make sense in the time formatting usecase but something like SECOND_OF_WEEK requires some bending over backwards.

**Comment by gianm** [comment]
@vogievetsky is there a standard or documentation for another database that describes the plyql TIME_PART function, or is it something special to plyql?

**Comment by vogievetsky** [comment]
Hi @gianm, I am sorry that I did not reply to you straight away. I felt that I needed to wait for the new year for my feelings on the subject to brew.

There is nothing like TIME_PART in general SQL as far as I know BUT the in most SQLs there are two things:

1. Lots and lots of basic time functions like (from: https://dev.mysql.com/doc/refman/5.7/en/date-and-time-functions.html ):
  - `MINUTE()` - MINUTE_OF_HOUR in my suggestion
  - `HOUR()` - HOUR_OF_DAY in my suggestion
  - `WEEKOFYEAR()` - WEEK_OF_YEAR in my suggestion
2. Ability to compose those very easily with Maths `HOUR(d)*60+MINUTE(d) AS "MINUTE_OF_DAY"`

In Plywood/PlyQL I thought it would be cool to lump all of these together into one function and then extend it will all the logical X_OF_Y. I did not want to ask for an extraction function for every function supported by MySQL but that would be fine as well. I figured that a good API would be to have a `timePart` extraction function with whatever subset of time parting functions you think makes sense.
